proto/cosmos/authz/v1beta1/tx.proto

```diff
@@ -40,7 +40,7 @@ message MsgGrant {
 
 // MsgExecResponse defines the Msg/MsgExecResponse response type.
 message MsgExecResponse {
-  cosmos.base.abci.v1beta1.Result result = 1;
+  repeated cosmos.base.abci.v1beta1.Result results = 1;
```

Comment:
Let's get rid of `Result` and just use `Any`. Result could be non deterministic because it contains log

Reply:
Or if we don't have the `Any` then just `repeated bytes` with the `data` field

Reply:
Would that affect https://github.com/cosmos/cosmos-sdk/pull/9522? I was planning on using the result field to emit the executed events

Reply:
We just need to emit them with the event emitter not in the result

Reply:
> Result could be non deterministic because it contains log

I thought the return value is not included in a block. 

> Or if we don't have the Any

IMHO, `Any` should be used only if we have a well defined interface.

Reply:
I wonder if it makes sens to include `events` as well - this way, other modules calling this module will be able to inspect events. WDYT?

Reply:
Data is included in the block but events and log are non deterministic. At least that's my understanding. So we should include just data and not events or log.

Reply:
I will change to `repeated byte responses` (where each response is the data attribute).

NOTE: I don't see where we add return values in a block. The `Data` in a Block is not response - it's a set of serialized transactions: https://github.com/tendermint/tendermint/blob/a6b30faf35fe28b30401e902bcf2498f092d2cf6/types/block.go#L1001
Header also doesn't have any commitment to responses nor events.

[digression]
So, since event commitment is not included in a block, then a state machines must not make any decision based on events fired by other service / module.  We can run a modified node with all events removed (all code triggering events) and will be still compatible with the rest of the network.

Reply:
Well events fired in the state machine are effectively deterministic, but from tendermint's perspective they aren't. Look at `abci.ResponseDeliverTx`